docs/source/en/package_reference/dataclasses.md

```diff
@@ -0,0 +1,215 @@
+# Strict Dataclasses
+
+The `huggingface_hub` package provides a utility to create **strict dataclasses**. These are enhanced versions of Python's standard `dataclass` with additional validation features. Strict dataclasses ensure that fields are validated both during initialization and assignment, making them ideal for scenarios where data integrity is critical.
+
+## Overview
+
+Strict dataclasses are created using the `@strict` decorator. They extend the functionality of regular dataclasses by:
+
+- Validating field types based on type hints
+- Supporting custom validators for additional checks
+- Optionally allowing arbitrary keyword arguments in the constructor
+- Validating fields both at initialization and during assignment
+
+## Benefits
+
+- **Data Integrity**: Ensures fields always contain valid data
+- **Ease of Use**: Integrates seamlessly with Python's `dataclass` module
+- **Flexibility**: Supports custom validators for complex validation logic
+- **Lightweight**: Requires no additional dependencies such as Pydantic, attrs, or similar libraries
+
+## Usage
+
+### Basic Example
+
+```python
+from dataclasses import dataclass
+from huggingface_hub.dataclasses import strict, as_validated_field
+
+# Custom validator to ensure a value is positive
+@as_validated_field
+def positive_int(value: int):
+    if not value >= 0:
```

Comment:
nit: positive numbers are `> 0` 🙈 (intentionally failing examples below set the value to `-1`, i.e. no examples are broken as a result of this change)

```suggestion
    if not value > 0:
```